components/new-contribution-flow/StepProfile.js

```diff
@@ -1,43 +1,348 @@
-import React from 'react';
+import React, { Fragment, useState } from 'react';
 import PropTypes from 'prop-types';
+import { InfoCircle } from '@styled-icons/boxicons-regular/InfoCircle';
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
 
-import { Box } from '../../components/Grid';
+import Avatar from '../../components/Avatar';
+import Container from '../../components/Container';
+import { Box, Flex } from '../../components/Grid';
+import InputTypeCountry from '../../components/InputTypeCountry';
+import Link from '../../components/Link';
+import MessageBox from '../../components/MessageBox';
+import StyledHr from '../../components/StyledHr';
+import StyledInput from '../../components/StyledInput';
+import StyledInputField from '../../components/StyledInputField';
+import StyledLink from '../../components/StyledLink';
+import StyledRadioList from '../../components/StyledRadioList';
+import StyledTextarea from '../../components/StyledTextarea';
 import { P } from '../../components/Text';
 import { withUser } from '../../components/UserProvider';
 
-class NewContributionFlowStepPayment extends React.Component {
-  static propTypes = {
-    collective: PropTypes.object,
-    LoggedInUser: PropTypes.object,
-    loadingLoggedInUser: PropTypes.object,
-    stepDetails: PropTypes.shape({
-      amount: PropTypes.number,
-      interval: PropTypes.string,
-    }),
+const messages = defineMessages({
+  incognito: { id: 'profile.incognito', defaultMessage: 'Incognito' },
+});
+
+const NewContributionFlowStepProfile = ({
+  LoggedInUser,
+  stepDetails,
+  profiles,
+  defaultSelectedProfile,
+  onChange,
+  data,
+}) => {
+  const intl = useIntl();
+  const [locale, setLocale] = useState('en');
+  const [localProfile, setLocalProfile] = useState(defaultSelectedProfile);
+
+  const renderInfoMessage = amount => {
```

Comment:
Defining multiple render functions in this component makes the function verbose. Can we extract that outside of this function, as a separate component like `<InfoMessage />`?